polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1029,3 +1037,122 @@ fn limit_and_sanitize_disputes<
 		(checked, checked_disputes_weight)
 	}
 }
+
+// Filters statements from disabled validators in `BackedCandidate` and `MultiDisputeStatementSet`.
+// Returns `true` if at least one statement is removed and `false` otherwise.
+fn filter_backed_statements<T: Config>(
```

Comment:
I just merged #1257 that has a `disabled_validators` implementation, which could be extracted into a common module. There's a small caveat though, it doesn't work at session boundaries. But maybe that's not really an issue since availability cores are cleared out on session boundaries anyway.

Reply:
I've moved your implementation in `pallet_shared` so that we can call it from the api impl and `paras_inherent`. Quite a lot of boilerplate code for such small change but it is what it is :)

Reply:
Btw if you have got an idea for a better place - please share.